test/test_runner.go

```diff
@@ -131,6 +132,8 @@ type TestRunner struct {
 	// the script environment, in order to aggregate and expose
 	// log messages from test cases and contracts.
 	logCollection *LogCollectionHook
+
+	backend *EmulatorBackend
```

Comment:
@m-Peter The `backend` here can be a problem similar to https://github.com/onflow/cadence-tools/pull/114#discussion_r1205952133.
We should probably fix this in a follow-up PR.

Reply:
Yes, I know :sweat_smile: I was caught off-guard, I didn't think it would be merged today :innocent: 
However, I did check the previous comment. Why do you think that it would be a problem? The `backend` is basically a singleton, it gets instantiated every time the `import Test` declaration is encountered, without actually calling `Test.newEmulatorBlockchain`. And the way the wiring is done in `flow-cli`, https://github.com/onflow/flow-cli/blob/master/internal/test/test.go#L120-L125, a new `TestRunner` is created per test file. Can you share a test case where a erroneous side-effect can manifest?

Reply:
Oh sorry, I just saw that you wrote `Maybe add a test case?` :sweat_smile: I will certainly add tests.

Reply:
yeah, I'm not also certain whether that 'is' actually a problem, but rather a 'hunch' that it may potentially cause problems.

A good example/test-case would be, in the same-test script:
- Create blockchain, `blockchain1`.
  - Deploy a contract `Foo` that emits an event `Foo.Event`.
  - Run a script/transaction that triggers the emitting of event `Foo.Event`.
  - Then get all events from `blockchain1`.
- Create another blockchain, `blockchain2`.
  - Deploy a contract `Bar` that emits an event `Bar.Event`.
  - Run a script/transaction that triggers the emitting of event `Bar.Event`.
  - Then get all events from `blockchain2`.

If this works fine, I guess we should be good.